src/query/groups.rs

```diff
@@ -51,3 +51,76 @@ impl Queryable<Authentication, crate::model::GroupAuditLogs>
 		}
 	}
 }
+
+/// Bans a user from a Group.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+pub struct GroupBan {
+	/// The ID of the group
+	pub group_id: crate::id::Group,
+	/// The ID of the user to ban
+	pub user_id: crate::id::User,
+}
+
+impl Queryable<Authentication, crate::model::GroupBan> for GroupBan {
+	fn url(&self, _: &Authentication) -> String {
+		format!("{}/groups/{}/bans", crate::API_BASE_URI, self.group_id.as_ref(),)
+	}
+
+	fn body(
+		&self, _state: &Authentication,
+	) -> Option<serde_json::Result<Vec<u8>>> {
+		Some(serde_json::to_vec(
+			&serde_json::json!({ "userId": self.user_id.as_ref() }),
+		))
+	}
+
+	fn method(&self, _state: &Authentication) -> racal::RequestMethod {
+		racal::RequestMethod::Post
+	}
+}
+
+/// Un-bans a user from a Group.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+pub struct GroupUnban {
+	/// The ID of the group
+	pub group_id: crate::id::Group,
+	/// The ID of the user to un-ban
+	pub user_id: crate::id::User,
+}
+
+impl Queryable<Authentication, crate::model::GroupBan> for GroupUnban {
+	fn url(&self, _: &Authentication) -> String {
+		format!(
+			"{}/groups/{}/bans/{}",
+			crate::API_BASE_URI,
+			self.group_id.as_ref(),
+			self.user_id.as_ref()
+		)
+	}
+
+	fn method(&self, _state: &Authentication) -> racal::RequestMethod {
+		racal::RequestMethod::Delete
+	}
+}
+
+/// Returns a Limited Group Member.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+pub struct GroupMember {
+	/// The ID of the group
+	pub group_id: crate::id::Group,
+	/// The ID of the user
+	pub user_id: crate::id::User,
+}
+
+impl Queryable<Authentication, Option<crate::model::GroupMember>>
+	for GroupMember
+{
+	fn url(&self, _: &Authentication) -> String {
+		format!(
+			"{}/groups/{}/members/{}",
+			crate::API_BASE_URI,
+			self.group_id.as_ref(),
+			self.user_id.as_ref()
+		)
+	}
+}
```

Comment:
I guess the other 2 queries can't really be sensibly tested, but this could use a live test as well to help quickly validate that the (de)serialization of the `GroupMember`s work at least :smile: